### DISCUSSION TITLE
How do I choose Database design for audit logging

### DISCUSSION THREAD

**Discussion by stockdaddy-dev** [nhost/nhost#2819] [Q&A]
The approaches that I know are:
1. Add columns for created and modified date and user
2. Insert only tables
3. Create a Separate history table for each table
4. Create a Consolidated history Table for All Tables

**Comment by dbarrosop** [comment]
that's going to depend a bit on requirements. If you don't have specific requirements other than tracking changes I'd suggest a global audit table + triggers as that's probably going to be the easiest solution.
